hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableMetaClient.java

```diff
@@ -117,6 +117,13 @@ public HoodieTableMetaClient(Configuration conf, String basePath, boolean loadAc
     TableNotFoundException.checkTableValidity(fs, basePathDir, metaPathDir);
     this.tableConfig = new HoodieTableConfig(fs, metaPath, payloadClassName);
     this.tableType = tableConfig.getTableType();
+    if (layoutVersion.isPresent()) {
+      // Ensure layout version passed in config is not lower than the one seen in hoodie.properties
+      TimelineLayoutVersion tableConfigVersion = tableConfig.getTimelineLayoutVersion();
```

Comment:
can't this be `null`? if so, would `compareTo` below be still happy?

Reply:
You are correct, Table Config Version can be null. I have made tableConfig.getTimelineLayoutVersion() to return an option instead